bundles/org.openhab.binding.nikohomecontrol/src/main/resources/OH-INF/addon/addon.xml

```diff
@@ -7,4 +7,18 @@
 	<name>@text/bindingName</name>
 	<description>@text/bindingDescription</description>
 	<connection>local</connection>
+
+	<discovery-methods>
+		<discovery-method>
+			<service-type>mdns</service-type>
+			<mdns-service-type>_http._tcp.local.</mdns-service-type>
+			<match-properties>
+				<match-property>
+					<name>name</name>
```

Comment:
I think it should be hostname, but didn’t test.

Reply:
I think if the fully qualified name is `foobar._http._tcp.local.` then the name part is`foobar`. (name.application.transport.domain). So when you speak of `hostName` are you talking about something other than `foobar`?

Reply:
Not sure. I took it from their documentation, and there it is called hostname. I guess I will have to try and confirm what the right pattern is.

Reply:
I tested and this is the correct version:
```
    "discoveryMethods": [
      {
        "serviceType": "mdns",
        "mdnsServiceType": "_http._tcp.local.",
        "matchProperties": [
          {
            "name": "hostname",
            "regex": "^(?i)FP00112A"
          }
        ]
      }
    ]
```

Reply:
It seems that in this case 'hostname' is not actually referring to the _"host name"_ of the device according to standard network parlance i.e. the `foobar` part of `foobar._http._tcp.local.` .. but rather, it appears to be device specific custom property called (confusingly) 'hostname'. => I will update the PR.

Reply:
Actually I think your `<regex>^(?i)FP00112A</regex>` is wrong (it matches almost anything). Probably should be `<regex>(?i)FP00112A.*</regex>` => ??

Reply:
On my system -- where there is definitely NOT a niko device -- with your code it was suggesting me to install that binding.

Reply:
I think there is a bug in the core mdns finder causing this: https://github.com/openhab/openhab-core/blob/fd11a996b4b6f09daf80c723d11320ad001f2f5c/bundles/org.openhab.core.config.discovery.addon/src/main/java/org/openhab/core/config/discovery/addon/finders/MDNSAddonSuggestionFinder.java#L90

If the property does not exist in the service, it is not validated. You need to check all properties effectively exist. I don’t think the problem is with the regex.

Reply:
So, correct version (and I now think this is final):
```
	<discovery-methods>
		<discovery-method>
			<service-type>mdns</service-type>
			<mdns-service-type>_http._tcp.local.</mdns-service-type>
			<match-properties>
				<match-property>
					<name>name</name>
					<regex>(?i)FP00112A.*</regex>
				</match-property>
			</match-properties>
		</discovery-method>
	</discovery-methods>
```

Reply:
Yup. I already committed that last night.

Reply:
Your commit still has `hostname` instead of `name` for the property.